src/CollAction/Helpers/ProjectBannerImageManager.cs

```diff
@@ -0,0 +1,68 @@
+﻿using CollAction.Models;
+using Microsoft.AspNetCore.Http;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace CollAction.Helpers
+{
+    public class ProjectBannerImageManager : IImageFileManager
+    {
+        public Project Project { get; set; }
+
+        private string WebFolder { get { return Path.Combine("usercontent", "bannerimages"); } }
+
+        // Don't trust the users _formFile.FileName to be unique, so generate our own unique one "banner_<Project.Id>.<FileType>"
```

Comment:
The comment doesn't match the code. I will change this comment to: ... own unique one "banner_<formatted-Project.Name>.<FileType>".